lib/cisco_node_utils/vlan.rb

```diff
@@ -199,5 +248,88 @@ def mapped_vni=(vni)
     def default_mapped_vni
       config_get_default('vlan', 'mapped_vni')
     end
+
+    def private_vlan_type
+      config_get('vlan', 'private_vlan_type', id: @vlan_id)
+    end
+
+    def private_vlan_type=(pv_type)
+      Feature.private_vlan_enable
+      fail TypeError unless pv_type && pv_type.is_a?(String)
+      set_args_keys_default
+      if pv_type == default_private_vlan_type
+        pv_type = private_vlan_type
+        @set_args[:vlan] = @vlan_id
+        @set_args[:state] = 'no'
+        @set_args[:type] = pv_type
+        ignore_msg = 'Warning: Private-VLAN CLI removed'
+      else
+        @set_args[:vlan] = @vlan_id
```

Comment:
Instead of setting `@set_args[:vlan] = @vlan_id` you can create the following two methods:
- https://github.com/cisco/cisco-network-node-utils/blob/b487026afcb19d5e180afc6756af7ce425db88d2/lib/cisco_node_utils/bgp_neighbor.rb#L80
- https://github.com/cisco/cisco-network-node-utils/blob/b487026afcb19d5e180afc6756af7ce425db88d2/lib/cisco_node_utils/bgp_neighbor.rb#L86

The fist method specifies the default keys used for each setter, and the second method resets the `@set_args` hash and then adds any needed property keys.

Here is an example of how you can call `set_args_keys` with the keys needed for the `config_set`